R/ADDIS.R

```diff
@@ -72,20 +73,31 @@
 #'
 #'
 #' @examples
-#' pval = c(2.90e-08, 0.06743, 3.51e-04, 0.0154, 0.04723,
-#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-06,
-#'         0.69274, 0.30443, 0.00136, 0.82342, 0.54757)
+#' sample.df <- data.frame(
+#' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' decision.times = seq_len(15) + 1)
 #'
-#' ADDIS(pval)
-#'
-#' ADDIS(pval, async=TRUE, decision.times=seq_len(15)) # Same as above
+#' ADDIS(sample.df, async = TRUE) # Asynchronous
 #' 
-#' ADDIS(pval, async=TRUE, decision.times=seq_len(15)+1) # Asynchronous
+#' ADDIS(sample.df, async = FALSE) # Synchronous
 #'
 #' @export
 
-ADDIS <- function(pval, alpha=0.05, gammai, w0, lambda=0.5, tau=0.5,
-                  async=FALSE, decision.times) {
+ADDIS <- function(d, alpha=0.05, gammai, w0, lambda=0.5, tau=0.5,
+                  async=FALSE) {
+    
+    if(is.data.frame(d)){
+        pval <- d$pval
+    } else if(is.vector(d)){
+        pval <- d
```

Comment:
Can add the following check here: if async = FALSE, then d needs to be a dataframe with a column of decision.times

Reply:
Done.